### DISCUSSION TITLE
Confusion about the query_embedding layer in DETR

### DISCUSSION THREAD

**Discussion by baibaidj** [open-mmlab/mmdetection#7023] [Q&A]
When I read the code of DETR, I found that the self.query_embedding was created as a nn.Embedding layer but only its weight was used in the model. I was a bit confused. See here: 
https://github.com/open-mmlab/mmdetection/blob/1a90fa80a761fe15e69111a625d82874ed783f7b/mmdet/models/dense_heads/detr_head.py#L259

If only the weight is needed, why not initialize the self.query_embedding by nn.Parameter(num_queries, embed_dims)?

**Comment by Li-Qingyun** [comment]
if it is initialized by nn.Parameter(num_queries, embed_dims)
it may need registry
